Retrying a single message will invoke all matching handlers again (#5386)
Particular/docs.particular.net: nservicebus/handlers/index.md

partial: behaviorcaveat

> ## Multiple handlers

Author: ```suggestion
## Invocation of multiple matching handlers
```

When an endpoint hosts multiple matching handlers for a single incoming message and one of the handlers fails then the incoming message gets retried. When the incoming message is retried all matching handlers get invoked again. Invoked handlers include the handlers that might had already been succesfully invoked in previous attempts.

In general, an incoming message represents a single unit of work. If multiple handlers get invoked it is best that either all succeed or all fail in a single transactional boundary. It is not recommended to run multiple atomic or isolated transactional operations for a single incoming message.

Ensure correct invocation of isolated or shared state by applying one of more of the strategies:

> - Ensuring all handlers share the same transactional context to ensure any the shared transaction can rollback any operations (connection sharing, MSDTC, etc.).

Reviewer: ```suggestion
- All handlers share the same transactional context to ensure that all participating operations rollback if any of them fails.
```

> - Ensuring handlers are idempotent and correctly deal with At-least-once Delivery.

Reviewer: ```suggestion
- Making handlers idempotent and correctly dealing with at-least-once delivery.
```

> - Hosting each handler in its own endpoint (hosting in isolation) so each subscribers receive its own own copy to process in isolation (multiple subscibers).

Reviewer: ```suggestion
- Hosting isolation - host each handler in its own endpoint so each subscriber receives its own copy of the message and processes it in isolation.
```

> - When receiving the message do a `SendLocal` with a new message but same payload so that each task gets represented by its own message and can be executed in isolation. (single subscriber, split at the target).

Reviewer: ```suggestion
- Split the message at the target - do `SendLocal` with a new message but the same payload so that each task gets represented by its own message and gets executed in isolation. 
```

> - Send individual commands from the endpoint that publishes the event, instead of a single event (split at the source).

Reviewer: ```suggestion
- Split the message at the source - send individual commands from the endpoint that publishes the event, instead of a single event.
```

> In general a generic message gets converted into several commands where each command represents a specific isolated task. Splitting in multiple message also has the benefit that each task can run in parallel where otherwise execution is sequential. 

Reviewer: ```suggestion
In general, a generic message gets converted into several commands where each command represents a single isolated task executed by a single handler. This approach has the additional benefit that each task can run in parallel where otherwise execution is sequential. 
```
